x-pack/plugins/cases/server/client/cases/get.ts

```diff
@@ -97,14 +117,43 @@ export const get = async ({
 /**
  * Retrieves the tags from all the cases.
  */
-export async function getTags({
-  savedObjectsClient: soClient,
-  caseService,
-  logger,
-}: CasesClientArgs): Promise<string[]> {
+
+export async function getTags(
+  params: AllTagsFindRequest,
+  clientArgs: CasesClientArgs
+): Promise<string[]> {
+  const {
+    savedObjectsClient: soClient,
+    caseService,
+    logger,
+    authorization: auth,
+    auditLogger,
+  } = clientArgs;
+
   try {
+    const queryParams = pipe(
+      excess(AllTagsFindRequestRt).decode(params),
+      fold(throwErrors(Boom.badRequest), identity)
+    );
+
+    let authFindHelpers: AuthorizationFilter;
+    try {
+      authFindHelpers = await auth.getFindAuthorizationFilter(CASE_SAVED_OBJECT);
+    } catch (error) {
+      auditLogger?.log(createAuditMsg({ operation: Operations.getTags, error }));
+      throw error;
+    }
+
+    const { filter: authorizationFilter } = authFindHelpers;
+    const queryArgs = {
+      owner: queryParams.owner,
+    };
+
+    const caseQueries = constructQueryOptions({ ...queryArgs, authorizationFilter });
```

Comment:
Hmm should we extract out the functionality for the `case.filter` so that it's a separate function used by `constructQueryOptions`? That way getTags and getReporters can call it directly instead of going through `constructQueryOptions`. I think I was imagining `constructQueryOptions` to only be called within the cases and sub cases find api. I named it poorly though 🤦‍♂️ .